### ISSUE TITLE
Check for autofocus doesn't respect dialog wrapping the element

### ISSUE THREAD

**Issue by AKlenner** [validator/validator#2057]
### Summary

When a page has two or more inputs with autofocus, the checker reports
> **Error**: There must not be two elements with the same "nearest ancestor autofocus scoping root element" that both have the autofocus attribute specified.

But according to the standard, multiple elements with autofocus are allowed, as long as they are nested in <dialog> or popover elements each.

### URL for document that reproduces the problem

n/a

### HTML/CSS/SVG snippet that reproduces the problem

```html
<!DOCTYPE html>
<html lang="">
<head>
<title>Test</title>
</head>
<body>

<input autofocus>

<dialog>
  <button aria-label="close dialog">X</button>
  <input autofocus>
</dialog>

</body>
</html>
```

### Checker error messages

```shell
Error: There must not be two elements with the same "nearest ancestor autofocus scoping root element" that both have the autofocus attribute specified.
```

### Spec links/citations

https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Global_attributes/autofocus

### Steps to reproduce

1. Run the code in the checker, f.e. in textbox

### Expected behavior

The checker should not throw an error for this code, as the elements with autofocus are separated by the dialog

### Actual behavior

The checker returns a misleading error.

**Comment by sideshowbarker** [comment]
Nice catch. Thanks for taking time to report it. Fixed now, in https://github.com/validator/validator/commit/65073343dd8441d3e2eb985c306cb98794763fbb — and deployed:

https://validator.w3.org/nu/?showsource=yes&doc=data%3Atext%2Fhtml%3Bcharset%3Dutf-8%3Bbase64%2CPCFET0NUWVBFIGh0bWw%2BCjxodG1sIGxhbmc9IiI%2BCjxoZWFkPgo8dGl0bGU%2BVGVzdDwvdGl0bGU%2BCjwvaGVhZD4KPGJvZHk%2BCgo8aW5wdXQgYXV0b2ZvY3VzPgoKPGRpYWxvZz4KICA8YnV0dG9uIGFyaWEtbGFiZWw9ImNsb3NlIGRpYWxvZyI%2BWDwvYnV0dG9uPgogIDxpbnB1dCBhdXRvZm9jdXM%2BCjwvZGlhbG9nPgoKPC9ib2R5Pgo8L2h0bWw%2B
